worlds/musedash/__init__.py

```diff
@@ -0,0 +1,289 @@
+    needed_token_count: int
+    location_count: int
+
+    def generate_early(self):
+        dlc_songs = self.multiworld.allow_just_as_planned_dlc_songs[self.player]
+        streamer_mode = self.multiworld.streamer_mode_enabled[self.player]
+
+        diff_threshold = self.get_difficulty_range()
+
+        available_song_keys = self.muse_dash_collection.get_songs_with_settings(dlc_songs, streamer_mode, diff_threshold[0], diff_threshold[1])
+        available_song_keys = self.handle_plando(available_song_keys)
+
+        self.create_song_pool(available_song_keys)
+
+        for song in self.starting_songs:
+            self.multiworld.push_precollected(self.create_item(song))
+
+    # Todo: Update this to list[str] when python 3.8 stops being used
+    def handle_plando(self, available_song_keys: list) -> list:
+        start_items = self.multiworld.start_inventory[self.player].value.keys()
+        include_songs = self.multiworld.include_songs[self.player].value
+        exclude_songs = self.multiworld.exclude_songs[self.player].value
+
+        self.starting_songs = [s for s in start_items if s in self.muse_dash_collection.song_items]
+        self.included_songs = [s for s in include_songs if (s in self.muse_dash_collection.song_items) and (s not in self.starting_songs)]
+        return [s for s in available_song_keys if (s not in start_items) and (s not in include_songs) and (s not in exclude_songs)]
+
+    # Todo: Update this to list[str] when python 3.8 stops being used
+    def create_song_pool(self, available_song_keys: list):
+        # Needs Starting Songs + Victory Song + At least 1 intermediary song
+        total_available_songs = len(available_song_keys)
+        startingSongCount = self.multiworld.starting_song_count[self.player]
+        if (total_available_songs < startingSongCount + 2):
+            raise Exception("Not enough songs available to satify the starting song count.")
+
+        self.victory_song_name = self.pop_random_item(available_song_keys)
+
+        for _ in range(0, startingSongCount):
+            self.starting_songs.append(self.pop_random_item(available_song_keys))
+
+        for _ in range(0, self.multiworld.additional_song_count[self.player]):
+            if (len(available_song_keys) <= 0):
+                break
+
+            self.included_songs.append(self.pop_random_item(available_song_keys))
+
+        self.location_count = len(self.starting_songs) + len(self.included_songs)
+        location_multiplier = 1 + (self.get_additional_item_percentage() / 100.0)
+        self.location_count = floor(self.location_count * location_multiplier)
+
+        minimum_location_count = len(self.included_songs) + self.get_music_sheet_count()
+        if (self.location_count < minimum_location_count):
+            self.location_count = minimum_location_count
+
+    # Todo: Update this to list[str] when python 3.8 stops being used
+    def pop_random_item(self, list: list) -> str:
```

Comment:
Pretty minor, but I think it's probably cleaner to remove this method and do

```python
self.multiworld.random.shuffle(my_list)

my_list.pop()
...
my_list.pop()
```

Reply:
Ah. Thanks for noticing this. Should now be fixed.